File: lib/models/report.dart
class Report {
  final String? id;
  final String driverLicense;
  final String route;
  final String complaint;
  final DateTime createdAt;
  final String status;

  Report({
    this.id,
    required this.driverLicense,
    required this.route,
    required this.complaint,
    DateTime? createdAt,
    this.status = 'pendiente',
  }) : createdAt = createdAt ?? DateTime.now();

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['_id'],
      driverLicense: json['driverLicense'],
      route: json['route'],
      complaint: json['complaint'],
      createdAt: DateTime.parse(json['createdAt']),
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'driverLicense': driverLicense,
      'route': route,
      'complaint': complaint,
      'createdAt': createdAt.toIso8601String(),
      'status': status,
    };
  }
} 

File: backend/src/models/report.js
const mongoose = require('mongoose');

const reportSchema = new mongoose.Schema({
  driverLicense: {
    type: String,
    required: true,
  },
  route: {
    type: String,
    required: true,
  },
  complaint: {
    type: String,
    required: true,
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  status: {
    type: String,
    enum: ['pendiente', 'en proceso', 'resuelto'],
    default: 'pendiente',
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Report', reportSchema); 